orchagent/bufferorch.cpp

```diff
@@ -645,6 +645,25 @@ task_process_status BufferOrch::processEgressBufferProfileList(Consumer &consume
     return task_process_status::task_success;
 }
 
+void BufferOrch::doTask()
+{
+    auto pool_consumer = getExecutor((CFG_BUFFER_POOL_TABLE_NAME));
+    pool_consumer->drain();
+
+    auto profile_consumer = getExecutor(CFG_BUFFER_PROFILE_TABLE_NAME);
+    profile_consumer->drain();
+
+    for(auto &it : m_consumerMap)
+    {
+        auto consumer = it.second.get();
+        if (consumer == profile_consumer)
+            continue;
+        if (consumer == pool_consumer)
+            continue;
+        consumer->drain();
+    }
```

Comment:
Do you mean the configuration for other buffer tables like CFG_BUFFER_QUEUE_TABLE_NAME, CFG_BUFFER_PG_TABLE_NAME, CFG_BUFFER_PORT_INGRESS_PROFILE_LIST_NAME and CFG_BUFFER_PORT_EGRESS_PROFILE_LIST_NAME have dependency on CFG_BUFFER_PROFILE_TABLE_NAME & CFG_BUFFER_POOL_TABLE_NAME?    CFG_BUFFER_PROFILE_TABLE_NAME has dependency on CFG_BUFFER_POOL_TABLE_NAME?

I didn't see existing code checking that dependency.

Reply:
Exactly, so I added comments. The ground truth is inside SAI design doc.
For code, task_process_status::task_need_retry possibly indicates a dependency missing. They don't show the dependency relationship as specific as SAI design doc.

---
In reply to: [212513136](https://github.com/Azure/sonic-swss/pull/590#discussion_r212513136) [](ancestors = 212513136)

Reply:
If there is dependency,  the exact dependency chain should be clarified and the problem should be fixed in normal code logic which applies to cold boot too.

Reply:
it is difficult to do this in the normal boot since the input sequence is unknown. we need to get the dependency at the entry level and then dispatched to task. in warmboot, since we have all the data, we can just solve the dependency at the table level.

Reply:
During cold start, the task_process_status::task_need_retry handles dependency correctly, ie. anything missing will postpone task processing. So there is no problem to fix.
I agree this code is not explicit in dependency relationship. This should be improved in future.

---
In reply to: [212527615](https://github.com/Azure/sonic-swss/pull/590#discussion_r212527615) [](ancestors = 212527615)